ports/stm32/can.c

```diff
@@ -219,16 +211,22 @@ int can_receive(CAN_HandleTypeDef *can, int fifo, CanRxMsgTypeDef *msg, uint8_t
     // Release (free) message from FIFO
     *rfr |= CAN_RF0R_RFOM0;
 
+    // Re-enable any interrupts that were disabled in RX IRQ handlers
+    can_enable_rx_interrupts(can, fifo, can_rx_pending(can, fifo) == 0);
```

Comment:
Note that in the original implementation (prior to this refactoring), RX interrupts (message pending, full, overflow) were only re-enabled if a Python callback was registered, eg `self->rxcallback0 != mp_const_none`.

But now, they are always re-enabled.

I guess this changes behaviour in a subtle way?  Because now the RX state machine (`rx_state0`, `rx_state1`) is managed even if there's no Python callback (prior, it was not managed).

Not sure if this matters too much, but worth thinking a bit more about it.

Reply:
As discussed, this is now delegated to the caller in pyb_can.c - which knows if interrupts are needed or not.